salt/modules/smbios.py

```diff
@@ -32,10 +33,16 @@
 DMIDECODER = salt.utils.path.which_bin(['dmidecode', 'smbios'])
 
 
+def _refresh_dmidecoder():
+    global DMIDECODER
+    DMIDECODER = salt.utils.path.which_bin(['dmidecode', 'smbios'])
```

Comment:
Why there is a need of this global variable at all? And this function is also not needed. I would discard this PR entirely and just refactor it the way that `which_bin` is just called.